CMakeLists.txt

```diff
@@ -50,6 +50,7 @@ if(USE_CUDA)
   message(STATUS "Use external CUB/Thrust library for a consistent API and performance.")
   cuda_include_directories(BEFORE "${CMAKE_SOURCE_DIR}/third_party/thrust")
   cuda_include_directories(BEFORE "${CMAKE_SOURCE_DIR}/third_party/thrust/dependencies/cub")
+  cuda_include_directories(BEFORE "${CMAKE_SOURCE_DIR}/third_party/thrust/dependencies/libcudacxx/include")
```

Comment:
Here introduces more dependencies on thrust. Could you try whether you can use the thrust library in CUDA?

Reply:
They come automatically when the thrust submodule in DGL is updated. New versions of thrust come with its own cub and libcudacxx.

Reply:
The CUB function I am using in this PR became part of CUB very recently. Thus, it will be available only with cuda 12. Since dgl supports cuda 11, DGL needs to continue to depend on thrust as a submodule.

Reply:
I see.

Reply:
Can you put a TODO here to remind us remove this once we deprecating CUDA 11?